.circleci/config.yml

```diff
@@ -337,63 +448,74 @@ jobs:
   deploy-staging:
     <<: *defaults
     steps:
-      - checkout
-      - run: *update_git_submodules
-      - restore_cache:
-          keys:
-          - api-cache-{{ checksum "~/workbench/api/build.gradle" }}
-          - api-cache-
+      - checkout_init_git
+      - manage_api_cache:
+          restore: true
+      - activate_service_account_creds
       - deploy:
           working_directory: ~/workbench/deploy
           command: |
-            ../ci/activate_creds.sh circle-sa-key.json
             ./project.rb deploy \
               --project all-of-us-rw-staging \
-              --account circle-deploy-account@all-of-us-workbench-test.iam.gserviceaccount.com \
+              --account << pipeline.parameters.circleci_service_account >> \
               --git-version "${CIRCLE_TAG}" \
               --app-version "${CIRCLE_TAG}" \
               --circle-url "https://circleci.com/workflow-run/${CIRCLE_WORKFLOW_ID}" \
               --key-file circle-sa-key.json \
               --promote
+          name: Deploy API and UI to "staging" App Engine
 
   deploy-perf:
     <<: *defaults
     steps:
-      - checkout
-      - run:
-          command: git submodule update --init --recursive
-      - restore_cache:
-          keys:
-          - api-cache-{{ checksum "~/workbench/api/build.gradle" }}
-          - api-cache-
+      - checkout_init_git
+      - manage_api_cache:
+          restore: true
+      - activate_service_account_creds
       - deploy:
-          name: Deploy to Perf Environment
           working_directory: ~/workbench/deploy
           command: |
-            ../ci/activate_creds.sh circle-sa-key.json
             ./project.rb deploy \
               --project all-of-us-rw-perf \
-              --account circle-deploy-account@all-of-us-workbench-test.iam.gserviceaccount.com \
+              --account << pipeline.parameters.circleci_service_account >> \
               --git-version "${CIRCLE_TAG}" \
               --app-version "${CIRCLE_TAG}" \
               --circle-url "https://circleci.com/workflow-run/${CIRCLE_WORKFLOW_ID}" \
               --key-file circle-sa-key.json \
               --promote
+          name: Deploy API and UI to "perf" App Engine
+
+  # Deploy local UI server and run Puppeteer 2e2 tests
+  puppeteer-e2e-local-ui:
```

Comment:
This is definitely a new job - should this be a separate PR?

ALso, I thought the plan was to run a local UI and API server. Did you try to get this working?

Reply:
Removed it from this PR. BTW, I couldn't get local ui/api working. I spent days on it, still having issues, I need your help!

Reply:
Sure thing, I didn't realize you were working on this. Please push up a draft PR of what you were trying and I can take a look.